### PR TITLE
Add black, flake8 and update the Travis CI with readme.

### PR DESCRIPTION
**Pull request by rexwangcc**
https://broadinstitute.atlassian.net/jira/software/projects/GH/boards/530

## Changes:
- [x] Add [Black](https://black.readthedocs.io/en/stable/index.html) as the code formatter for this repo.
- [x] Add Flake-8 and its configuration to the repo.
- [x] Update the Travis CI config so we have a linting test before all of the other tests.
- [x] Add `pre-commit` and update the requirements as well as the readme.
- [x] Add some status badges.
 

## Review+QA Instructions:

- **Please expect the test to fail until we merge the PR 586 into this feature branch!!**
- Please check the new multi-stage test on Travis, especially verify that there is a new "linting test" before all tests.
- Please follow the updated readme, make sure the `pre-commit` is installed on your local environment and if feasible, try to make a commit (e.g. add an empty line to the readme, but don't push it), see if the hook works. **Using a virtualenv is highly recommended!**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,20 +1,66 @@
...
+
+- [Skylab](https://github.com/HumanCellAtlas/skylab): Standardized HCA data processing pipelines
+
+## Development
+
+### File Structure Layout
+
+_Note: The following layout is generated by `tree -L 2 -d`_
+```
+.
```
Comment (samanehsan) [inline README.md]: Can we remove this `.` here?

File: README.md [hunk 2]
```diff
@@ -1,20 +1,66 @@
...
+
+- [Skylab Analysis](https://github.com/HumanCellAtlas/skylab-analysis): Analysis and benchmarking reports for standardized HCA pipelines
+
+- [Skylab](https://github.com/HumanCellAtlas/skylab): Standardized HCA data processing pipelines
+
+## Development
+
+### File Structure Layout
+
+_Note: The following layout is generated by `tree -L 2 -d`_
```
Comment (samanehsan) [inline README.md]: Include how the layout was generated doesn't seem like necessary information here

Comment (rexwangcc) [reply README.md]: removed!

File: README.md (hunk 1 above)
Comment (rexwangcc) [reply README.md]: Sure!

Comment (rexwangcc) [reply README.md]: Actually removing it results in weird layout above certs, going to add slash after the `.`

**Pull request merged**